internal/planner/consumer_test.go

```diff
@@ -0,0 +1,888 @@
+/*
+ * Copyright 2025 Canonical Ltd.
+ * See LICENSE file for licensing details.
+ *
+ * Unit tests for the consumer.
+ */
+
+package planner
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"maps"
+	"testing"
+	"time"
+
+	"github.com/canonical/github-runner-operators/internal/database"
+	"github.com/canonical/github-runner-operators/internal/telemetry"
+	amqp "github.com/rabbitmq/amqp091-go"
+	"github.com/stretchr/testify/assert"
+)
+
+type fakeDB struct {
+	jobs              map[string]*database.Job
+	addErr            error
+	updateStartErr    error
+	updateCompleteErr error
+}
+
+func newFakeDB() *fakeDB {
+	return &fakeDB{jobs: make(map[string]*database.Job)}
+}
+
+func (db *fakeDB) AddJob(ctx context.Context, job *database.Job) error {
+	if db.addErr != nil {
+		return db.addErr
+	}
+
+	key := jobKey(job.Platform, job.ID)
+	if _, exists := db.jobs[key]; exists {
+		return database.ErrExist
+	}
+	db.jobs[key] = job
+	return nil
+}
+
+func (db *fakeDB) UpdateJobStarted(ctx context.Context, platform, id string, startedAt time.Time, raw map[string]any) error {
+	if db.updateStartErr != nil {
+		return db.updateStartErr
+	}
+
+	key := jobKey(platform, id)
+	job, exists := db.jobs[key]
+	if !exists {
+		return database.ErrNotExist
+	}
+	job.StartedAt = &startedAt
+	mergeRaw(job, raw)
+	return nil
+}
+
+func (db *fakeDB) UpdateJobCompleted(ctx context.Context, platform, id string, completedAt time.Time, raw map[string]any) error {
+	if db.updateCompleteErr != nil {
+		return db.updateCompleteErr
+	}
+	key := jobKey(platform, id)
+	job, exists := db.jobs[key]
+	if !exists {
+		return database.ErrNotExist
+	}
+	job.CompletedAt = &completedAt
+	mergeRaw(job, raw)
+	return nil
+}
+
+func mergeRaw(job *database.Job, raw map[string]interface{}) {
+	if raw == nil {
+		return
+	}
+	if job.Raw == nil {
+		job.Raw = raw
+		return
+	}
+	maps.Copy(job.Raw, raw)
+}
+
+func jobKey(platform, id string) string { return platform + ":" + id }
+
+// fakeAmqpConsumer mocks the queue.AmqpConsumer for testing JobConsumer
+type fakeAmqpConsumer struct {
+	deliveries <-chan amqp.Delivery
+	pullErr    error
+}
+
+func (c *fakeAmqpConsumer) Pull(ctx context.Context) (amqp.Delivery, error) {
+	if c.pullErr != nil {
+		return amqp.Delivery{}, c.pullErr
+	}
+
+	select {
+	case msg, ok := <-c.deliveries:
+		if !ok {
+			// Channel closed, wait for context to be done
+			<-ctx.Done()
+			return amqp.Delivery{}, ctx.Err()
+		}
+		return msg, nil
+	case <-ctx.Done():
+		return amqp.Delivery{}, ctx.Err()
+	}
+}
+
+func TestConsumer(t *testing.T) {
+	mk := func(m map[string]any) []byte { b, _ := json.Marshal(m); return b }
```

Comment:
can you add the arrange/act/assert comment as advised in the STYLE guide https://github.com/canonical/github-runner-operators/blob/main/STYLE.md#comments-in-tests